nBTC/sources/redeem_request.move

```diff
@@ -166,8 +170,15 @@ public fun raw_signed_tx(r: &RedeemRequest, storage: &Storage): vector<u8> {
 }
 
 // add valid signature to redeem request for specify input index
-public(package) fun add_signature(r: &mut RedeemRequest, input_idx: u32, signature: vector<u8>) {
-    r.signatures_map.insert(input_idx, signature);
+public(package) fun add_signature(
+    r: &mut RedeemRequest,
+    input_idx: u32,
+    ika_signature: vector<u8>,
+) {
+    // TODO: ika signature for ECDSA alway return 65 bytes length
+    assert!(ika_signature.length() == 65, EInvalidIkaECDSALength);
+    let raw_signature = ika_signature.slice(1, 65); // skip the first element or recover id
```

Comment:
i dont understand the comment. What do you mean by `or recover id`

Reply:
```suggestion
    let raw_signature = ika_signature.slice(1, 65); // skip the first byte (pub key recovery byte)
```